save-preprocessor/src/main/kotlin/com/saveourtool/save/preprocessor/service/GitPreprocessorService.kt

```diff
@@ -0,0 +1,126 @@
+package com.saveourtool.save.preprocessor.service
+
+import com.saveourtool.save.entities.GitDto
+import com.saveourtool.save.preprocessor.config.ConfigProperties
+import com.saveourtool.save.preprocessor.utils.cloneToDirectory
+import com.saveourtool.save.utils.*
+import org.eclipse.jgit.util.FileUtils
+import org.slf4j.Logger
+import org.springframework.stereotype.Service
+import reactor.core.publisher.Mono
+import reactor.core.scheduler.Schedulers
+import java.io.IOException
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.time.Instant
+import kotlin.io.path.absolutePathString
+import kotlin.io.path.createDirectories
+
+typealias GitRepositoryProcessor<T> = (Path, Instant) -> Mono<T>
+typealias ArchiveProcessor<T> = (Path) -> Mono<T>
+
+/**
+ * Additional service for Git based [com.saveourtool.save.entities.TestSuitesSource]s
```

Comment:
Don't get - why do we need this Service? 
For example: `cloneAndProcessDirectory` - I thought we already had such logic in preprocessor. Or how we download repositories?

Reply:
I extracted it to a separate Service to has access to configProperties and create a tmp folder